src/reflection.ts

```diff
@@ -697,4 +728,172 @@ export class Parser {
       return result;
     };
   }
+
+  readVectorOfUnionsLambda(
+    field: reflection.Field,
+    discriminatorField: reflection.Field,
+    readDefaults: boolean,
+  ): (table: Table) => Record<string, any>[] {
+    const fieldType = field.type();
+    if (fieldType === null) {
+      throw new Error(`Malformed schema: "type" field of '${field.name()}' not populated.`);
+    }
+
+    if (discriminatorField.type()?.baseType() !== reflection.BaseType.Vector) {
+      throw new Error(`Malformed schema: union discriminator field is not a vector`);
+    }
+
+    // This will deserialize the vector of discriminator values
+    const readDiscriminators = this.readVectorOfScalarsLambda(discriminatorField);
+
+    // This will read the vector of tables from the actual field
+    // How to deserialize each table will depend on the discriminator value at the same vector
+    // index position in the discriminator values field
+    const vectorLambda = this.readVectorOfTablesLambda(field);
+
+    const unionDeserializers = this.createUnionDeserializers(fieldType, readDefaults);
+
+    return (table: Table) => {
+      const discriminators = readDiscriminators(table);
+      if (!discriminators) {
+        throw new Error(
+          `Malformed message: missing vector discriminators field: ${discriminatorField.name()}`,
+        );
+      }
+
+      const tables = vectorLambda(table);
+      if (tables == null) {
+        throw new Error(`Malformed message: missing vector table for field: ${field.name()}`);
+      }
+
+      if (discriminators.length !== tables.length) {
+        throw new Error(
+          `malformed message: ${field.name()} length != ${discriminatorField.name()} length`,
+        );
+      }
+
+      const result = [];
+      for (let idx = 0; idx < tables.length; ++idx) {
+        const discriminator = discriminators[idx];
+        if (typeof discriminator !== "number") {
+          throw new Error(
+            `Malformed union discriminator value is not a number in field: ${discriminatorField.name()}`,
+          );
+        }
+
+        // Skip NONE
```

Comment:
(a) do you have any test-cases covering this NONE case; and
(b) should we push an `undefined` or `null` or something to the vector instead of skipping the value?

Reply:
I actually couldn't identify that it was valid to have NONE cases in the vector. The generated type script types did not allow me to push `undefined` or `null` without a type error. Do you know if this is a thing you can or would do in the generated c++? Maybe there's a way to build a "NONE" instance that I am not aware of?

Reply:
I'll take a look to see if it seems possible. A comment to that effect would be helpful for when we come back later and question this in the future.